services/custom-forms.ts

```diff
@@ -127,11 +176,69 @@ export async function updateField(
     linkedFormTypeTtl = `${escaped.fieldUri} ext:linkedFormType ${linkedFormTypeUri} .`;
   }
 
+  const libraryEntryData = await getLibraryEntryForField(
+    description.field,
+    description.libraryEntryUri,
+  );
+  let fieldPath = null;
+  let libraryEntryInsertTtl = '';
+  let removeFieldPathWhen = '';
+  let insertFieldPathWhen = '';
+  if (libraryEntryData) {
+    fieldPath = libraryEntryData.fieldPath;
+    const _escaped = {
+      libraryEntry: sparqlEscapeUri(libraryEntryData.libraryEntryUri),
+      validation: libraryEntryData.validationUri
+        ? sparqlEscapeUri(libraryEntryData.validationUri)
+        : null,
+      path: sparqlEscapeUri(libraryEntryData.fieldPath),
+    };
+    let validation = '';
+    if (libraryEntryData.validationUri) {
+      validation = `
+        ${escaped.fieldUri} form:validatedBy ${_escaped.validation} .
+        ${_escaped.validation} sh:path ${_escaped.path} .
+      `;
+    }
+    libraryEntryInsertTtl = `
+      ${escaped.fieldUri} prov:wasDerivedFrom ${_escaped.libraryEntry} .
+      ${escaped.fieldUri} sh:path ${_escaped.path} .
+      ${validation}
+    `;
+  }
+
+  const currentFieldTypeIsLibraryEntry = await isFieldOfTypeLibraryEntry(
```

Comment:
don't we know from the libraryEntryDate being null or not?

Reply:
ah no! there is a difference between the old value and the new value. maybe we should call them that? `oldFieldIsLibraryEntry` and `newFieldLibraryEntryData`?